protocols/kad/src/handler.rs

```diff
@@ -364,6 +364,15 @@ pub struct KademliaRequestId {
     connec_unique_id: UniqueConnecId,
 }
 
+#[cfg(test)]
+impl Default for KademliaRequestId {
```

Comment:
As long as `KademliaRequestId` implements `Clone`, this `Default` impl might as well be available all the time.

Reply:
This is an opaque implementation detail. Implementing `Default` on it doesn't make any sense. Maybe it could contain a channel or an `Arc` in the future.

Reply:
I understood `KademliaRequestId` to be some sort of unique one-time token and `Clone` defeats that purpose but I did not look much into it so feel free to ignore the criticism.

Reply:
I am fine either way. I can't come up with a use-case for `Default` outside of testing though.

Reply:
> I understood `KademliaRequestId` to be some sort of unique one-time token and `Clone` defeats that purpose but I did not look much into it so feel free to ignore the criticism.

Just to give some context: `KademliaRequestId` did not use to be `Clone` but it had to be as a result of https://github.com/libp2p/rust-libp2p/pull/1440 requiring `Clone` for the `TInEvent` of any handler, since it is now possible to "broadcast" a single such event to multiple handlers (which in turn was a requirement discovered during the substrate integration in https://github.com/paritytech/substrate/pull/5066) and `KademliaRequestId` is part of these events in `libp2p-kad`. It would of course be better to not have that `Clone`, but I didn't come up with a way to do so.